tests/edgeless.spec.ts

```diff
@@ -341,3 +340,93 @@ test('hovering on shape should not have effect on underlying block', async ({
   await page.mouse.move(x + 50, y + 50);
   await assertEdgelessHoverRect(page, [x, y, 100, 100]);
 });
+
+test('pan tool basic', async ({ page }) => {
+  await enterPlaygroundRoom(page);
+  await initEmptyEdgelessState(page);
+  await switchEditorMode(page);
+
+  const start = { x: 100, y: 100 };
+  const end = { x: 200, y: 200 };
+  await addBasicRectShapeElement(page, start, end);
+
+  await setMouseMode(page, 'pan');
+  await dragBetweenCoords(
+    page,
+    {
+      x: start.x + 5,
+      y: start.y + 5,
+    },
+    {
+      x: start.x + 25,
+      y: start.y + 25,
+    }
+  );
+  await setMouseMode(page, 'default');
+
+  await page.mouse.move(start.x + 25, start.y + 25);
+  await assertEdgelessHoverRect(page, [120, 120, 100, 100]);
+});
+
+test('pan tool shortcut', async ({ page }) => {
+  await enterPlaygroundRoom(page);
+  await initEmptyEdgelessState(page);
+  await switchEditorMode(page);
+
+  const start = { x: 100, y: 100 };
+  const end = { x: 200, y: 200 };
+  await addBasicRectShapeElement(page, start, end);
+
+  await page.mouse.move(start.x + 5, start.y + 5);
+  await assertEdgelessHoverRect(page, [100, 100, 100, 100]);
+
+  await page.keyboard.down('Space');
+  const panButton = page
```

Comment:
Better use an action to query pan button, seems to be reusable in the case below.